pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go

```diff
@@ -190,6 +234,48 @@ type AlertRuleGroup struct {
 	Rules     []ProvisionedAlertRule `json:"rules"`
 }
 
+// AlertRuleFileExport is the provisioned export of multiple models.AlertRuleGroup.
+// swagger:model
+type AlertRuleFileExport struct {
+	Groups []AlertRuleGroupExport `json:"groups" yaml:"groups"`
+}
+
+// AlertRuleGroupExport is the provisioned export of models.AlertRuleGroup.
+// swagger:model
+type AlertRuleGroupExport struct {
```

Comment:
This struct is identical to the struct that file provisioning reads in:
https://github.com/grafana/grafana/blob/a77d95807ca0e85dd2745a2f5ba2795ddef44c49/pkg/services/provisioning/alerting/rules_types.go#L26

Can we use one single definition of this rather than creating it twice?

The `services/provisioning/alerting` package is allowed to depend on ngalert packages  (it already depends on models) we could move it here instead of duplicating.

Same goes for the other structs below.

Reply:
I agree this would be better in theory. I spent some time trying to get this to work but eventually noticed that this would mean using the various `values.{String,Int64,...}Value` in the API. This could:
-  Potentially expose their unmarshalling functionality of expanding os env variables. Now, I think we could do this safely by only allowing their use in responses not in inputs but it still gives me pause as it might not be immediately obvious to future devs that we do such a dangerous operation as env expansion directly in the unmarshaller.
- (less important) We can no longer `omitempty` these fields without some annoying null Pointer "hacks".
 
After some discussion offline, I'm going to keep the separate definitions but instead try to define these structs closer to each other to help reduce likelihood of drift.